test_engine/terraform/aws/ubuntu/main.tf

```diff
@@ -0,0 +1,321 @@
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "Allow UDP connection for longhorn-webhooks"
+    from_port   = 0
+    to_port     = 65535
+    protocol    = "udp"
+    cidr_blocks = ["10.0.0.0/8"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "build_engine_aws_sec_grp_controlplane-${random_string.random_suffix.id}"
+  }
+}
+
+
+# Create worker security group
+resource "aws_security_group" "build_engine_aws_secgrp_worker" {
+  name        = "build_engine_aws_secgrp_worker"
+  description = "Allow all inbound traffic"
+  vpc_id      = aws_vpc.build_engine_aws_vpc.id
+
+  ingress {
+    description = "Allow All Traffic from VPC CIDR block"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = [aws_vpc.build_engine_aws_vpc.cidr_block]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "build_engine_aws_sec_grp_worker-${random_string.random_suffix.id}"
+  }
+}
+
+
+# Create Public subnet
+resource "aws_subnet" "build_engine_aws_public_subnet" {
+  vpc_id     = aws_vpc.build_engine_aws_vpc.id
+  availability_zone = var.aws_availability_zone
+  cidr_block = "10.0.1.0/24"
+
+  tags = {
+    Name = "build_engine_public_subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create private subnet
+resource "aws_subnet" "build_engine_aws_private_subnet" {
+  vpc_id     = aws_vpc.build_engine_aws_vpc.id
+  availability_zone = var.aws_availability_zone
+  cidr_block = "10.0.2.0/24"
+
+  tags = {
+    Name = "build_engine_private_subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create EIP for NATGW
+resource "aws_eip" "build_engine_aws_eip_nat_gw" {
+  vpc      = true
+
+  tags = {
+    Name = "build_engine_eip_nat_gw-${random_string.random_suffix.id}"
+  }
+}
+
+# Create nat gateway
+resource "aws_nat_gateway" "build_engine_aws_nat_gw" {
+  depends_on = [
+    aws_internet_gateway.build_engine_aws_igw,
+    aws_eip.build_engine_aws_eip_nat_gw,
+    aws_subnet.build_engine_aws_public_subnet,
+    aws_subnet.build_engine_aws_private_subnet
+  ]
+
+  allocation_id = aws_eip.build_engine_aws_eip_nat_gw.id
+  subnet_id     = aws_subnet.build_engine_aws_public_subnet.id
+
+  tags = {
+    Name = "build_engine_eip_nat_gw-${random_string.random_suffix.id}"
+  }
+}
+
+
+# Create route table for public subnets
+resource "aws_route_table" "build_engine_aws_public_rt" {
+  depends_on = [
+    aws_internet_gateway.build_engine_aws_igw,
+  ]
+
+  vpc_id = aws_vpc.build_engine_aws_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.build_engine_aws_igw.id
+  }
+
+  tags = {
+    Name = "build_engine_aws_public_rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Create route table for private subnets
+resource "aws_route_table" "build_engine_aws_private_rt" {
```

Comment:
we don't need a router table for private subnet

Reply:
I had modified this part of code